### PR TITLE
Add support of Renesas Salvator XS M3 and Spider boards

### PR DESCRIPTION
**Pull request by xakep-amatop**
Add support of Renesas Salvator XS M3 and Spider boards

### CODE REVIEW

File: dts/arm64/renesas/gen3_common/r8a7795_r8a7796_common.dtsi [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+#include <arm64/armv8-a.dtsi>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <zephyr/dt-bindings/clock/renesas_cpg_mssr.h>
+#include <zephyr/dt-bindings/clock/r8a7795_cpg_mssr.h>
+
+/ {
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	cpus {
```
Comment (lorc) [inline dts/arm64/renesas/gen3_common/r8a7795_r8a7796_common.dtsi]: It is not good idea to put `cpus` node into common DTSi. H3 and M3 has different set of CPUs.

File: dts/arm64/renesas/gen3_common/r8a7795_r8a7796_common.dtsi [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		<GIC_PPI 10 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>;
+	};
+
+	gic: interrupt-controller@f1010000 {
+		compatible = "arm,gic-400", "arm,gic" ;
+		#interrupt-cells = <4>;
+		#address-cells = <0>;
+		interrupt-controller;
+		reg = <0 0xf1010000 0 0x1000>,
+			<0 0xf1020000 0 0x20000>;
```
Comment (lorc) [inline dts/arm64/renesas/gen3_common/r8a7795_r8a7796_common.dtsi]: Strange formatting.

File: dts/arm64/renesas/gen3_common/r8a7795_r8a7796_common.dtsi [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	psci {
+		compatible = "arm,psci-1.0", "arm,psci-0.2", "arm,psci";
+		method = "smc";
+	};
+
+	arch_timer: timer {
+		compatible = "arm,armv8-timer";
+		interrupt-parent = <&gic>;
+		interrupts = <GIC_PPI 13 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>,
+		<GIC_PPI 14 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>,
```
Comment (lorc) [inline dts/arm64/renesas/gen3_common/r8a7795_r8a7796_common.dtsi]: Strange formatting.

File: boards/arm64/renesas_rcar/gen3/common_salvator_xs_h3ulcb.dts [hunk 4]
```diff
@@ -1,18 +1,12 @@
 /*
- * Copyright (c) 2019 Carlo Caione <ccaione@baylibre.com>
+ * Copyright (c) 2023 EPAM Systems
```
Comment (lorc) [inline boards/arm64/renesas_rcar/gen3/common_salvator_xs_h3ulcb.dts]: Ugh. This is gross. Never ever remove  other's copyright notes.

Also newer put our copyright notes on a copied code  (of course, if original code was written by not us).

You can add a second copyright note if you did major changes to the file, though

File: drivers/clock_control/CMakeLists.txt [hunk 5]
```diff
@@ -53,9 +53,10 @@ zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_AGILEX clock_agilex.c)
 
 if(CONFIG_CLOCK_CONTROL_RCAR_CPG_MSSR)
   zephyr_library_sources(clock_control_renesas_cpg_mssr.c)
-if(CONFIG_SOC_R8A77951 OR CONFIG_SOC_R8A7795 OR CONFIG_SOC_R8A7796)
-  zephyr_library_sources(clock_control_r8a7795_cpg_mssr.c)
-endif()
+  if(CONFIG_SOC_R8A77951 OR CONFIG_SOC_R8A7795 OR CONFIG_SOC_R8A7796)
```
Comment (lorc) [inline drivers/clock_control/CMakeLists.txt]: This is getting out of control. I believe, you need to add new Kconfig option, like `CONFIG_SOC_CONFIG_SOC_R8A779x` and make conditional compilation based on it.

File: dts/arm64/renesas/r8a779f.dtsi [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+	psci {
+		compatible = "arm,psci-0.2";
+		method = "smc";
+	};
+
+	arch_timer: timer {
+		compatible = "arm,armv8-timer";
+		interrupt-parent = <&gic>;
+		interrupts = <GIC_PPI 13 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>,
+		<GIC_PPI 14 IRQ_TYPE_LEVEL IRQ_DEFAULT_PRIORITY>,
```
Comment (lorc) [inline dts/arm64/renesas/r8a779f.dtsi]: Strange formatting

File: boards/shields/xenvm_dom0/boards/rcar_spider.overlay [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/delete-node/ &ram;
+
+/ {
+	ram: memory@60000000 {
+		device_type = "mmio-sram";
+		reg = <0x00 0x60000000 0x00 DT_SIZE_M(512)>;
```
Comment (lorc) [inline boards/shields/xenvm_dom0/boards/rcar_spider.overlay]: Did you considered making it configurable via Kconfig?

File: boards/shields/xenvm_dom0/boards/rcar_salvator_xs_m3_2x4g.overlay [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/delete-node/ &ram;
+/delete-node/ &scif2;
+
+/ {
+	ram: memory@60000000 {
+		device_type = "mmio-sram";
+		reg = <0x00 0x60000000 0x00 DT_SIZE_M(512)>;
```
Comment (lorc) [inline boards/shields/xenvm_dom0/boards/rcar_salvator_xs_m3_2x4g.overlay:13]: Did you considered making it configurable via Kconfig?

**Review by lorc — COMMENTED** [review]
It is "bare-metal", not "bar-metal". Please fix your commit description.

File: boards/shields/xenvm_dom0/boards/rcar_salvator_xs_m3_2x4g.overlay (hunk 8 above)
Comment (xakep-amatop) [reply boards/shields/xenvm_dom0/boards/rcar_salvator_xs_m3_2x4g.overlay:13]: as far as we can use both dts overlays and Kconfig-s for configuring of every board, I see no reason to add it Kconfig
moreover, I thought that we would use some mechanism for modifying dts

File: dts/arm64/renesas/gen3_common/r8a7795_r8a7796_common.dtsi (hunk 1 above)
Comment (xakep-amatop) [reply dts/arm64/renesas/gen3_common/r8a7795_r8a7796_common.dtsi]: as far as know Renesas gen3 m3 and h3 boards use the same CPUs

File: drivers/clock_control/CMakeLists.txt [hunk 9]
```diff
@@ -53,9 +53,8 @@ zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_AGILEX clock_agilex.c)
 
 if(CONFIG_CLOCK_CONTROL_RCAR_CPG_MSSR)
   zephyr_library_sources(clock_control_renesas_cpg_mssr.c)
-if(CONFIG_SOC_R8A77951 OR CONFIG_SOC_R8A7795)
-  zephyr_library_sources(clock_control_r8a7795_cpg_mssr.c)
-endif()
+  zephyr_library_sources_ifdef(CONFIG_DT_HAS_RENESAS_R8A7795_CPG_MSSR_ENABLED clock_control_r8a7795_cpg_mssr.c)
+  zephyr_library_sources_ifdef(CONFIG_SOC_R8A779F clock_control_r8a779f_cpg_mssr.c)
```
Comment (lorc) [inline drivers/clock_control/CMakeLists.txt]: This looks inconsistent. Why `r8a7795` depends on auto-generated  DTS entry, but  `r8A779f` - on Kconfig option from SOC?

File: dts/arm64/renesas/gen3_common/r8a7795_r8a7796_common.dtsi (hunk 1 above)
Comment (lorc) [reply dts/arm64/renesas/gen3_common/r8a7795_r8a7796_common.dtsi]: Yes and no. There are the same CPU models, but there are different count of CPUs. As I said, H3 and M3 has different **sets** of CPUs. Putting `cpus` node into common part will cause problems later, when we'll enable SMP and will be required to provide actual CPUs configuration.

File: boards/shields/xenvm_dom0/boards/rcar_salvator_xs_m3_2x4g.overlay (hunk 8 above)
Comment (lorc) [reply boards/shields/xenvm_dom0/boards/rcar_salvator_xs_m3_2x4g.overlay:13]: Fair enough

File: dts/arm64/renesas/gen3_common/r8a7795_r8a7796_common.dtsi (hunk 1 above)
Comment (xakep-amatop) [reply dts/arm64/renesas/gen3_common/r8a7795_r8a7796_common.dtsi]: done

File: drivers/clock_control/CMakeLists.txt (hunk 9 above)
Comment (xakep-amatop) [reply drivers/clock_control/CMakeLists.txt]: done

**Comment by lorc** [comment]
`Acked-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>`

File: soc/arm64/renesas_rcar/gen4/soc.h [hunk 10]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) 2021 IoT.bzh
```
Comment (firscity) [inline soc/arm64/renesas_rcar/gen4/soc.h]: Probably copy/paste issue, don't think it is related to Gen4 (also in 2021)

**Review by firscity — COMMENTED** [review]
`Acked-by: Dmytro Firsov <dmytro_firsov@epam.com>`

**Pull request merged**